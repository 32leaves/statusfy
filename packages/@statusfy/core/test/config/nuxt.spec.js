import path, { isAbsolute } from 'path'
import isString from 'lodash.isstring'
import cloneDeepWith from 'lodash.clonedeepwith'

import generateConfig from '@/core/lib/config/generate'
const tempPath = path.resolve(__dirname, '../.tmp')

const relative = (dir) => {
  return path.relative(__dirname, dir)
}

describe('config:nuxt', () => {
  const executeConfigTest = label => {
    const sourceDir = path.resolve(tempPath, label)

    const generatedConfig = generateConfig(sourceDir, {})
    let nuxtConfig = generatedConfig.nuxtConfig

    // Check absolute dirs
    expect(isAbsolute(nuxtConfig.buildDir)).toBeTruthy()
    expect(isAbsolute(nuxtConfig.generate.dir)).toBeTruthy()
    expect(isAbsolute(nuxtConfig.icon.iconSrc)).toBeTruthy()
    expect(isAbsolute(nuxtConfig.rootDir)).toBeTruthy()
    // expect(isAbsolute(nuxtConfig.srcDir)).toBeTruthy()
    expect(isAbsolute(nuxtConfig.statusfy.assets.mainLogo)).toBeTruthy()
    if (nuxtConfig.statusfy.publicFilesPath) {
      expect(isAbsolute(nuxtConfig.statusfy.publicFilesPath)).toBeTruthy()
    }
    expect(isAbsolute(nuxtConfig.statusfy.siteConfig.sourceDir)).toBeTruthy()
    expect(isAbsolute(nuxtConfig.workbox.globDirectory)).toBeTruthy()
    for (const dir of nuxtConfig.modulesDir) {
      expect(isString(dir)).toBeTruthy()
    }

    expect(nuxtConfig.modulesDir.length).toBeGreaterThanOrEqual(2)

    // Convert absolute paths to relative
    nuxtConfig = cloneDeepWith(nuxtConfig, (value) => {
      if (isString(value) && isAbsolute(value)) {
        const relativePath = path.normalize(relative(value))

        if (relativePath.split('../').length <= 3) {
          return path.normalize(relative(value))
        }
      }
    })

    // Check config
    expect(nuxtConfig).toMatchSnapshot()
  }

  test('simple', () => {
    executeConfigTest('simple')
  })

  test('cli', () => {
    executeConfigTest('cli')
  })

  test('advanced', () => {
    executeConfigTest('advanced')
  })
})
