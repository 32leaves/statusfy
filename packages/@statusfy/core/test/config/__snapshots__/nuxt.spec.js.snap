// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config:nuxt advanced 1`] = `
Object {
  "axios": Object {
    "browserBaseURL": "/",
  },
  "build": Object {
    "analyze": true,
    "extend": [Function],
    "plugins": Array [
      Object {
        "apply": [Function],
      },
    ],
    "publicPath": "/static/",
    "router": Object {
      "linkActiveClass": "active",
      "linkExactActiveClass": "exact",
      "scrollBehavior": [Function],
    },
  },
  "buildDir": "../.tmp/advanced/.statusfy",
  "css": Array [
    "~assets/css/tailwind.css",
    "@fortawesome/fontawesome-svg-core/styles.css",
    "../.tmp/advanced/theme/default/style.css",
    "../.tmp/advanced/theme/default/style.less",
    "../.tmp/advanced/theme/default/style.sass",
    "../.tmp/advanced/theme/default/style.scss",
    "../.tmp/advanced/theme/default/style.styl",
    "../.tmp/advanced/theme/default/style.stylus",
  ],
  "dev": true,
  "env": Object {
    "isDev": true,
  },
  "generate": Object {
    "dir": "../.tmp/advanced/dist",
    "fallback": "404.html",
    "subFolders": false,
  },
  "head": Object {
    "link": Array [
      Object {
        "color": "#3e4e88",
        "href": "/icons/safari-pinned-tab.svg",
        "rel": "mask-icon",
      },
    ],
    "meta": Array [
      Object {
        "charset": "utf-8",
      },
      Object {
        "content": "width=device-width, initial-scale=1",
        "name": "viewport",
      },
      Object {
        "content": "Statusfy",
        "name": "generator",
      },
      Object {
        "content": "",
        "hid": "description",
        "name": "description",
      },
      Object {
        "content": "",
        "hid": "description",
        "name": "description",
      },
      Object {
        "content": "Description",
        "hid": "description",
        "name": "description",
      },
    ],
    "title": "Title",
    "titleTemplate": "%s | Title",
  },
  "icon": Object {
    "iconSrc": "../.tmp/advanced/theme/default/img/icon.png",
    "sizes": Array [
      16,
      120,
      144,
      152,
      192,
      384,
      512,
    ],
  },
  "loading": Object {
    "color": "#1b1f23",
  },
  "manifest": Object {
    "description": "Description",
    "lang": "en",
    "name": "Title",
    "publicPath": "/static/",
    "short_name": "Title",
    "start_url": "/?standalone=true&utm_source=web_app&utm_medium=pwa",
    "version": "0.1.0-alpha.4",
  },
  "meta": Object {
    "author": "Statusfy",
    "charset": "utf-8",
    "mobileApp": false,
    "mobileAppIOS": false,
    "ogDescription": false,
    "ogHost": "https://demo.statusfy.co",
    "ogTitle": false,
    "theme_color": "#1b1f23",
  },
  "mode": "universal",
  "modules": Array [
    "~/modules/statusfy",
    "@nuxtjs/axios",
    Array [
      "nuxt-i18n",
      Object {
        "baseUrl": "https://demo.statusfy.co",
        "defaultLocale": "en",
        "detectBrowserLanguage": Object {
          "cookieKey": "title.lang_redirected",
          "useCookie": true,
        },
        "langDir": "locales/",
        "lazy": true,
        "locales": Array [
          Object {
            "code": "en",
            "file": "en.js",
            "iso": "en-US",
            "name": "English",
          },
          Object {
            "code": "en",
            "file": "en.js",
            "iso": "en-US",
            "name": "English",
          },
          Object {
            "code": "en",
            "file": "en.js",
            "iso": "en-US",
            "name": "English",
          },
          Object {
            "code": "es",
            "file": "es.js",
            "iso": "es-ES",
            "name": "Español",
          },
        ],
        "vueI18n": Object {
          "fallbackLocale": "en",
          "silentTranslationWarn": false,
        },
      },
    ],
    Array [
      "@nuxtjs/pwa",
      Object {
        "manifest": false,
        "workbox": false,
      },
    ],
  ],
  "modulesDir": Array [
    "../../node_modules",
    "../../../node_modules",
    "../.tmp/simple/node_modules",
    "../.tmp/cli/node_modules",
    "../.tmp/advanced/node_modules",
  ],
  "plugins": Array [
    "~/plugins/axios",
    "~/plugins/vue-svgicon",
  ],
  "rootDir": "../..",
  "server": Object {
    "host": "127.0.0.1",
    "port": 3000,
  },
  "srcDir": "../../client",
  "statusfy": Object {
    "assets": Object {
      "mainLogo": "../.tmp/advanced/theme/default/img/logo.svg",
    },
    "dev": true,
    "iconSizes": Array [
      16,
      120,
      144,
      152,
      192,
      384,
      512,
    ],
    "langDir": "locales/",
    "locales": Array [
      Object {
        "code": "en",
        "file": "en.json",
        "iso": "en-US",
        "name": "English",
      },
      Object {
        "code": "es",
        "file": "es.json",
        "iso": "es-ES",
        "name": "Español",
      },
    ],
    "publicFilesPath": "../.tmp/advanced/public",
    "siteConfig": Object {
      "analytics": Object {
        "ga": "UA-XXXXXXXXX-Y",
      },
      "baseUrl": "https://demo.statusfy.co",
      "build": Object {
        "isStatic": false,
      },
      "content": Object {
        "dir": "content",
        "frontMatterFormat": "yaml",
        "systems": Array [
          "cdn",
          "conversions",
          "site-delivery",
          "api",
        ],
      },
      "defaultLocale": "en",
      "description": "Description",
      "head": Object {
        "link": Array [
          Object {
            "color": "#3e4e88",
            "href": "/icons/safari-pinned-tab.svg",
            "rel": "mask-icon",
          },
        ],
        "meta": Array [
          Object {
            "charset": "utf-8",
          },
          Object {
            "content": "width=device-width, initial-scale=1",
            "name": "viewport",
          },
          Object {
            "content": "Statusfy",
            "name": "generator",
          },
          Object {
            "content": "",
            "hid": "description",
            "name": "description",
          },
          Object {
            "content": "",
            "hid": "description",
            "name": "description",
          },
          Object {
            "content": "Description",
            "hid": "description",
            "name": "description",
          },
        ],
        "title": "Title",
        "titleTemplate": "%s | Title",
      },
      "locales": Array [
        Object {
          "code": "en",
          "file": "en.json",
          "iso": "en-US",
          "name": "English",
        },
        Object {
          "code": "es",
          "file": "es.json",
          "iso": "es-ES",
          "name": "Español",
        },
      ],
      "manifest": true,
      "name": "title",
      "serviceWorker": true,
      "sourceDir": "../.tmp/advanced",
      "title": "Title",
    },
    "sourceDir": "../.tmp/advanced",
    "version": "0.1.0-alpha.4",
  },
  "workbox": Object {
    "_runtimeCaching": Array [
      Object {
        "handler": "cacheFirst",
        "method": "GET",
        "urlPattern": "/static/(?!content).*$",
      },
    ],
    "cacheId": "title",
    "globDirectory": "../.tmp/advanced/.statusfy/dist/client",
    "offlinePage": "/offline",
    "publicPath": "/static/",
    "runtimeCaching": Array [
      Object {
        "strategyOptions": Object {
          "cacheExpiration": Object {
            "maxAgeSeconds": 300,
            "maxEntries": 10,
          },
          "cacheName": "title_title_api",
        },
        "urlPattern": "(/static/content)?/api/v0/.*",
      },
    ],
  },
}
`;

exports[`config:nuxt cli 1`] = `
Object {
  "axios": Object {
    "browserBaseURL": "/",
  },
  "build": Object {
    "analyze": true,
    "extend": [Function],
    "plugins": Array [
      Object {
        "apply": [Function],
      },
    ],
    "publicPath": "/static/",
    "router": Object {
      "linkActiveClass": "active",
      "linkExactActiveClass": "exact",
      "scrollBehavior": [Function],
    },
  },
  "buildDir": "../.tmp/cli/.statusfy",
  "css": Array [
    "~assets/css/tailwind.css",
    "@fortawesome/fontawesome-svg-core/styles.css",
  ],
  "dev": true,
  "env": Object {
    "isDev": true,
  },
  "generate": Object {
    "dir": "../.tmp/cli/dist",
    "fallback": "404.html",
    "subFolders": false,
  },
  "head": Object {
    "meta": Array [
      Object {
        "charset": "utf-8",
      },
      Object {
        "content": "width=device-width, initial-scale=1",
        "name": "viewport",
      },
      Object {
        "content": "Statusfy",
        "name": "generator",
      },
      Object {
        "content": "",
        "hid": "description",
        "name": "description",
      },
      Object {
        "content": "",
        "hid": "description",
        "name": "description",
      },
    ],
    "title": "Title",
    "titleTemplate": "%s | Title",
  },
  "icon": Object {
    "iconSrc": "../../client/static/icon.png",
    "sizes": Array [
      16,
      120,
      144,
      152,
      192,
      384,
      512,
    ],
  },
  "loading": Object {
    "color": "#1b1f23",
  },
  "manifest": Object {
    "description": "",
    "lang": "en",
    "name": "Title",
    "publicPath": "/static/",
    "short_name": "Title",
    "start_url": "/?standalone=true&utm_source=web_app&utm_medium=pwa",
    "version": "0.1.0-alpha.4",
  },
  "meta": Object {
    "author": "Statusfy",
    "charset": "utf-8",
    "mobileApp": false,
    "mobileAppIOS": false,
    "ogDescription": false,
    "ogHost": "/",
    "ogTitle": false,
    "theme_color": "#1b1f23",
  },
  "mode": "universal",
  "modules": Array [
    "~/modules/statusfy",
    "@nuxtjs/axios",
    Array [
      "nuxt-i18n",
      Object {
        "baseUrl": null,
        "defaultLocale": "en",
        "detectBrowserLanguage": Object {
          "cookieKey": "title.lang_redirected",
          "useCookie": true,
        },
        "langDir": "locales/",
        "lazy": true,
        "locales": Array [
          Object {
            "code": "en",
            "file": "en.js",
            "iso": "en-US",
            "name": "English",
          },
          Object {
            "code": "en",
            "file": "en.js",
            "iso": "en-US",
            "name": "English",
          },
        ],
        "vueI18n": Object {
          "fallbackLocale": "en",
          "silentTranslationWarn": false,
        },
      },
    ],
    Array [
      "@nuxtjs/pwa",
      Object {
        "manifest": false,
        "workbox": false,
      },
    ],
  ],
  "modulesDir": Array [
    "../../node_modules",
    "../../../node_modules",
    "../.tmp/simple/node_modules",
    "../.tmp/cli/node_modules",
  ],
  "plugins": Array [
    "~/plugins/axios",
    "~/plugins/vue-svgicon",
  ],
  "rootDir": "../..",
  "server": Object {
    "host": "127.0.0.1",
    "port": 3000,
  },
  "srcDir": "../../client",
  "statusfy": Object {
    "assets": Object {
      "mainLogo": "../../client/assets/img/logo.svg",
    },
    "dev": true,
    "iconSizes": Array [
      16,
      120,
      144,
      152,
      192,
      384,
      512,
    ],
    "langDir": "locales/",
    "locales": Array [
      Object {
        "code": "en",
        "file": "en.json",
        "iso": "en-US",
        "name": "English",
      },
    ],
    "publicFilesPath": "../.tmp/cli/public",
    "siteConfig": Object {
      "baseUrl": "/",
      "build": Object {
        "isStatic": false,
      },
      "content": Object {
        "dir": "content",
        "systems": Array [
          "api",
          "cdn",
          "dns",
          "site-delivery",
        ],
      },
      "defaultLocale": "en",
      "description": "",
      "locales": Array [
        Object {
          "code": "en",
          "file": "en.json",
          "iso": "en-US",
          "name": "English",
        },
      ],
      "manifest": true,
      "name": "title",
      "serviceWorker": true,
      "sourceDir": "../.tmp/cli",
      "title": "Title",
    },
    "sourceDir": "../.tmp/cli",
    "version": "0.1.0-alpha.4",
  },
  "workbox": Object {
    "_runtimeCaching": Array [
      Object {
        "handler": "cacheFirst",
        "method": "GET",
        "urlPattern": "/static/(?!content).*$",
      },
    ],
    "cacheId": "title",
    "globDirectory": "../.tmp/cli/.statusfy/dist/client",
    "offlinePage": "/offline",
    "publicPath": "/static/",
    "runtimeCaching": Array [
      Object {
        "strategyOptions": Object {
          "cacheExpiration": Object {
            "maxAgeSeconds": 300,
            "maxEntries": 10,
          },
          "cacheName": "title_api",
        },
        "urlPattern": "(/static/content)?/api/v0/.*",
      },
    ],
  },
}
`;

exports[`config:nuxt simple 1`] = `
Object {
  "axios": Object {
    "browserBaseURL": "/",
  },
  "build": Object {
    "analyze": true,
    "extend": [Function],
    "plugins": Array [
      Object {
        "apply": [Function],
      },
    ],
    "publicPath": "/static/",
    "router": Object {
      "linkActiveClass": "active",
      "linkExactActiveClass": "exact",
      "scrollBehavior": [Function],
    },
  },
  "buildDir": "../.tmp/simple/.statusfy",
  "css": Array [
    "~assets/css/tailwind.css",
    "@fortawesome/fontawesome-svg-core/styles.css",
  ],
  "dev": true,
  "env": Object {
    "isDev": true,
  },
  "generate": Object {
    "dir": "../.tmp/simple/dist",
    "fallback": "404.html",
    "subFolders": false,
  },
  "head": Object {
    "meta": Array [
      Object {
        "charset": "utf-8",
      },
      Object {
        "content": "width=device-width, initial-scale=1",
        "name": "viewport",
      },
      Object {
        "content": "Statusfy",
        "name": "generator",
      },
      Object {
        "content": "",
        "hid": "description",
        "name": "description",
      },
    ],
    "title": "Title",
    "titleTemplate": "%s | Title",
  },
  "icon": Object {
    "iconSrc": "../../client/static/icon.png",
    "sizes": Array [
      16,
      120,
      144,
      152,
      192,
      384,
      512,
    ],
  },
  "loading": Object {
    "color": "#1b1f23",
  },
  "meta": Object {
    "author": "Statusfy",
    "charset": "utf-8",
    "mobileApp": false,
    "mobileAppIOS": false,
    "ogDescription": false,
    "ogHost": "/",
    "ogTitle": false,
    "theme_color": "#1b1f23",
  },
  "mode": "universal",
  "modules": Array [
    "~/modules/statusfy",
    "@nuxtjs/axios",
    Array [
      "nuxt-i18n",
      Object {
        "baseUrl": null,
        "defaultLocale": "en",
        "detectBrowserLanguage": Object {
          "cookieKey": "title.lang_redirected",
          "useCookie": true,
        },
        "langDir": "locales/",
        "lazy": true,
        "locales": Array [
          Object {
            "code": "en",
            "file": "en.js",
            "iso": "en-US",
            "name": "English",
          },
        ],
        "vueI18n": Object {
          "fallbackLocale": "en",
          "silentTranslationWarn": false,
        },
      },
    ],
    Array [
      "@nuxtjs/pwa",
      Object {
        "manifest": false,
        "workbox": false,
      },
    ],
  ],
  "modulesDir": Array [
    "../../node_modules",
    "../../../node_modules",
    "../.tmp/simple/node_modules",
  ],
  "plugins": Array [
    "~/plugins/axios",
    "~/plugins/vue-svgicon",
  ],
  "rootDir": "../..",
  "server": Object {
    "host": "127.0.0.1",
    "port": 3000,
  },
  "srcDir": "../../client",
  "statusfy": Object {
    "assets": Object {
      "mainLogo": "../../client/assets/img/logo.svg",
    },
    "dev": true,
    "iconSizes": Array [
      16,
      120,
      144,
      152,
      192,
      384,
      512,
    ],
    "langDir": "locales/",
    "locales": Array [
      Object {
        "code": "en",
        "iso": "en-US",
        "name": "English",
      },
    ],
    "publicFilesPath": null,
    "siteConfig": Object {
      "baseUrl": "/",
      "build": Object {
        "isStatic": false,
      },
      "content": Object {
        "dir": "content",
        "systems": Array [
          "api",
          "cdn",
          "dns",
          "site-delivery",
        ],
      },
      "defaultLocale": "en",
      "description": "",
      "locales": Array [
        Object {
          "code": "en",
          "iso": "en-US",
          "name": "English",
        },
      ],
      "manifest": false,
      "name": "title",
      "serviceWorker": false,
      "sourceDir": "../.tmp/simple",
      "title": "Title",
    },
    "sourceDir": "../.tmp/simple",
    "version": "0.1.0-alpha.4",
  },
}
`;
